packages/terra-clinical-application-site/src/demo/panel-manager/PanelManager.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SlidePanel from 'terra-slide-panel';
+import SlideGroup from 'terra-clinical-slide-group';
+import AppDelegate from 'terra-clinical-app-delegate';
+
+const propTypes = {
+  app: AppDelegate.propType,
+  children: PropTypes.node,
+  panelComponentData: PropTypes.array,
+  size: PropTypes.string,
+  isOpen: PropTypes.bool,
+  isMaximized: PropTypes.bool,
+
+  openPanel: PropTypes.func,
+  closePanel: PropTypes.func,
+  pushPanel: PropTypes.func,
+  popPanel: PropTypes.func,
+  maximizePanel: PropTypes.func,
+  minimizePanel: PropTypes.func,
+};
+
+const defaultProps = {
+  isOpen: false,
+  isMaximized: false,
+  size: 'small',
+  panelComponentData: [],
+};
+
+class PanelManager extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.buildPanelComponents = this.buildPanelComponents.bind(this);
+    this.buildChildren = this.buildChildren.bind(this);
+  }
+
+  buildPanelComponents() {
+    const { app, panelComponentData, isMaximized, pushPanel, popPanel, closePanel, maximizePanel, minimizePanel } = this.props;
+
+    return panelComponentData.map((componentData, index) => {
+      const ComponentClass = AppDelegate.getComponentForDisclosure(componentData.name);
+
+      if (!ComponentClass) {
+        return undefined;
+      }
+
+      const appDelegate = AppDelegate.create({
+        disclose: (data) => {
+          if (data.preferredType === 'panel' || !app) {
+            pushPanel(data);
+          } else {
+            app.disclose(data);
+          }
+        },
+        dismiss: (index > 0 ?
+          () => {
+            popPanel();
+          } :
+          () => {
+            closePanel();
+          }
+        ),
+        closeDisclosure: () => { closePanel(); },
+        goBack: index > 0 ? () => { popPanel(); } : null,
+        maximize: !isMaximized ? () => { maximizePanel(); } : null,
+        minimize: isMaximized ? () => { minimizePanel(); } : null,
+      });
+
+      return <ComponentClass key={componentData.key} {...componentData.props} app={appDelegate} />;
```

Comment:
That's pretty neat that `ComponentClass = AppDelegate.getComponentForDisclosure(componentData.name);` this will render a component!